src/test/rustdoc/macro_pub_in_module.rs

```diff
@@ -0,0 +1,49 @@
+// aux-build:macro_pub_in_module.rs
+// edition:2018
+// build-aux-docs
+// @has external_crate/some_module/macro.external_macro.html
+
+//! See issue #74355
+#![feature(decl_macro, no_core, rustc_attrs)]
+#![crate_name = "krate"]
+#![no_core]
+
+extern crate external_crate;
+
+pub mod inner {
+    // @has krate/inner/macro.raw_const.html
+    pub macro raw_const() {}
+
+    // @has krate/inner/macro.test.html
+    #[rustc_builtin_macro]
+    pub macro test($item:item) {}
+
+    // @has krate/inner/macro.Clone.html
+    #[rustc_builtin_macro]
+    pub macro Clone($item:item) {}
+
+    // Make sure the logic is not affected by a re-export.
+    mod private {
+        pub macro m() {}
+    }
+    // @has krate/inner/macro.renamed.html
+    pub use private::m as renamed;
```

Comment:
What happens if you rename the module, not the macro?
```suggestion
    // Make sure the logic is not affected by a re-export.
    mod private {
        pub macro m() {}
    }
    // @has krate/inner/macro.renamed.html
    pub use private::m as renamed;
    
    mod private2 {
        pub macro m2() {}
    }
    use private2 as renamed_mod;
    // @has krate/inner/macro.m2.html
    pub use renamed_mod::m2;
```